homeassistant/components/insteon_plm.py

```diff
@@ -6,115 +6,176 @@
 """
 import logging
 import asyncio
+import collections
 
 import voluptuous as vol
 
 from homeassistant.core import callback
-from homeassistant.const import (
-    CONF_PORT, EVENT_HOMEASSISTANT_STOP)
+from homeassistant.const import (CONF_PORT, EVENT_HOMEASSISTANT_STOP)
 import homeassistant.helpers.config_validation as cv
 from homeassistant.helpers import discovery
 
-REQUIREMENTS = ['insteonplm==0.7.5']
+REQUIREMENTS = ['insteonplm==0.8.2']
 
 _LOGGER = logging.getLogger(__name__)
 
 DOMAIN = 'insteon_plm'
 
 CONF_OVERRIDE = 'device_override'
+CONF_NEWNAMES = 'new_names'
 
 CONFIG_SCHEMA = vol.Schema({
     DOMAIN: vol.Schema({
         vol.Required(CONF_PORT): cv.string,
-        vol.Optional(CONF_OVERRIDE, default=[]): cv.ensure_list_csv,
-    })
+        vol.Optional(CONF_OVERRIDE, default=[]): vol.All(
+            cv.ensure_list_csv, vol.Length(min=1)),
+        vol.Optional(CONF_NEWNAMES, default=['no']): cv.string
+        })
 }, extra=vol.ALLOW_EXTRA)
 
-PLM_PLATFORMS = {
-    'binary_sensor': ['binary_sensor'],
-    'light': ['light'],
-    'switch': ['switch'],
-}
-
 
 @asyncio.coroutine
 def async_setup(hass, config):
     """Set up the connection to the PLM."""
     import insteonplm
 
+    ipdb = IPDB()
+
     conf = config[DOMAIN]
     port = conf.get(CONF_PORT)
     overrides = conf.get(CONF_OVERRIDE)
+    newnames = conf.get(CONF_NEWNAMES)
+
+    if isinstance(newnames, list):
+        newnames = ''
+    use_newnames = False
+    if newnames.lower() == 'y' or newnames.lower() == 'yes':
+        use_newnames = True
 
     @callback
     def async_plm_new_device(device):
         """Detect device from transport to be delegated to platform."""
-        name = device.get('address')
-        address = device.get('address_hex')
-        capabilities = device.get('capabilities', [])
-
-        _LOGGER.info("New INSTEON PLM device: %s (%s) %r",
-                     name, address, capabilities)
-
-        loadlist = []
-        for platform in PLM_PLATFORMS:
-            caplist = PLM_PLATFORMS.get(platform)
-            for key in capabilities:
-                if key in caplist:
-                    loadlist.append(platform)
-
-        loadlist = sorted(set(loadlist))
-
-        for loadplatform in loadlist:
-            hass.async_add_job(
-                discovery.async_load_platform(
-                    hass, loadplatform, DOMAIN, discovered=[device],
-                    hass_config=config))
-
+        _LOGGER = logging.getLogger(__name__)
```

Comment:
Don't get a logger each time. Instead add it to top of the page.

Reply:
Done